util/reggen/data.py

```diff
@@ -158,8 +157,8 @@ def get_nested_dims(self):
         # or if this is the last multiregister level in a nested multiregister
         if not isinstance(self, MultiReg):
             dims = [len(self.get_fields_flat())]
-        if  isinstance(self, MultiReg) and   \
-            not isinstance(self.fields[0], MultiReg):
+        if isinstance(self, MultiReg) and\
+           not isinstance(self.fields[0], MultiReg):
```

Comment:
I understand that flake8 no longer moans about this, but it feels like this is a cunning way to trick the tool, rather than a clean solution. My suggestion of
``` python
        if (isinstance(self, MultiReg) and
            not isinstance(self.fields[0], MultiReg)):
            if self.ishomog:
                dims = [len(self.get_fields_flat())]
            else:
                dims = [len(self.fields)]
```
causes a warning from flake8 because the contents of the `if` test line up with the body. And, actually, looking at the code, I think I agree with the tool. This is just bad luck: "if" has two characters. My usual approach is to add another layer of parentheses:
```python
        if ((isinstance(self, MultiReg) and
             not isinstance(self.fields[0], MultiReg))):
            if self.ishomog:
                dims = [len(self.get_fields_flat())]
            else:
                dims = [len(self.fields)]
```

Reply:
Hum, I think it is also awkward to add two parentheses without any reason. I prefer no parentheses but having one pair of parentheses is okay.